docs/studies/hidden_linear_function.ipynb

```
@@ -0,0 +1,540 @@
+    "\n",
+    "Code below can be used to generate random problem with given size of $\\mathcal{L}_q$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Size of subspace L: 4\n",
+      "Number of solutions: 256\n"
+     ]
+    }
+   ],
+   "source": [
+    "def random_problem(n, seed=None):\n",
+    "    \"\"\"Generates instance of the problem with given `n`.\n",
+    "    \n",
+    "    Args:\n",
+    "        n: dimension of the problem.\n",
+    "    \"\"\"\n",
+    "    if seed is not None:\n",
+    "        np.random.seed(seed) \n",
+    "    A = np.random.randint(0, 2, size=(n,n))\n",
+    "    for i in range(n):\n",
+    "        for j in range(i+1):\n",
+    "            A[i][j] = 0\n",
+    "    b = np.random.randint(0, 2, size=n)\n",
+    "    problem = HiddenLinearFunctionProblem(A, b)\n",
+    "    return problem\n",
+    "        \n",
+    "def find_interesting_problem(n, min_L_size):\n",
+    "    \"\"\"Generates \"interesting\" instance of the problem.\n",
+    "    \n",
+    "    Args:\n",
+    "        n: dimension of the problem.\n",
+    "        min_L_size: minimal cardinality of subspace L.\n",
+    "        \n",
+    "    Returns instance of problem with gien `n`, such that size of \n",
+    "    subspace `L_q` is at least `min_L_size`.\"\"\"\n",
+    "    for _ in range(1000):\n",
+    "        problem = random_problem(n)\n",
+    "        problem.bruteforce_solve()\n",
+    "        if len(problem.L) >= min_L_size and not np.max(problem.A) == 0:\n",
+    "            return problem\n",
+    "    return None\n",
```

nit:  Should we raise an error instead of silently returning None?

---

This is auxillary code, so I don't think it matters.
And, strictly speaking, it's valid (not exceptional) situation if you didn't find "interesting" problem - for example, if min_L_size is to high.